### PR TITLE
Parallelize renameRecursively for distcp.

### PR DESCRIPTION
**Pull request by ibuenros**

### CODE REVIEW

**Comment by ibuenros** [comment]
@pcadabam can you review?

File: gobblin-utility/src/main/java/gobblin/util/deprecation/DeprecationUtils.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   * Sets an option in a {@link State} to the first available value of a list of deprecatedKeys. For example, if
+   * an option "optiona" was previously called "optionb" or "optionc",
+   * calling {@link #handleDeprecatedOptions(State, String, List)} will search for the first available key-value pair
+   * with key optiona, optionb, or optionc, and set optiona to that value.
+   *
+   * @param state {@link State} to modify.
+   * @param currentKey current name of an option.
+   * @param deprecatedKeys all other names that {@link State}s could use to refer to that option.
+   */
+  public static final void handleDeprecatedOptions(State state, String currentKey, List<String> deprecatedKeys) {
```
Comment (pcadabam) [inline gobblin-utility/src/main/java/gobblin/util/deprecation/DeprecationUtils.java]: `migrateDeprecatedKeys` or `renameDeprecatedKeys`?

File: gobblin-utility/src/main/java/gobblin/util/filesystem/FileSystemUtils.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import gobblin.util.Decorator;
+import gobblin.util.DecoratorUtils;
+import gobblin.util.RateControlledFileSystem;
+import gobblin.util.deprecation.DeprecationUtils;
+
+
+/**
+ * Utilities for {@link FileSystem}s.
+ */
+public class FileSystemUtils {
```
Comment (pcadabam) [inline gobblin-utility/src/main/java/gobblin/util/filesystem/FileSystemUtils.java]: Most of the FileSystem utilities are in `HadoopUtils`. Can you move it there?

**Comment by pcadabam** [comment]
LGTM. Couple of of minor refactoring changes.

**Pull request merged**